### PR TITLE
validates before saving or updating a time_request

### PR DESCRIPTION
**Pull request by MounirGaiby**

### CODE REVIEW

**Comment by MounirGaiby** [comment]
[fix time requests bug when submiting empty dates](https://trello.com/c/MSF65YkD/310-fix-time-requests-bug-when-submiting-empty-dates)

File: app/controllers/time_requests_controller.rb [hunk 1]
```diff
@@ -60,25 +60,29 @@ def edit
...
-      render turbo_stream: [
-        turbo_stream.replace("pagination", partial: "shared/pagination", locals: {records: @time_requests}),
-        turbo_stream.prepend("records-list", partial: "time_requests/time_request", locals: {record: @time_request}),
-        turbo_stream.replace("time_off_balance", partial: "time_requests/time_off_balance", locals: {user_id: @time_request.user_request.user_id}),
-        turbo_stream.replace("right", partial: "shared/right"),
-        turbo_stream.replace("notification_alert", partial: "layouts/alert")
-      ]
+    if @time_request.valid?
+      @time_request.duration = new_time_request_duration(time_request_params, @user.id, @user.function.schedule)
+      if @time_request.save
```
Comment (MehdiElrafi) [inline app/controllers/time_requests_controller.rb:70]: I think you don't need the this if here, you can save it directly

File: app/controllers/time_requests_controller.rb [hunk 2]
```diff
@@ -60,25 +60,29 @@ def edit
...
+        @pto_taken_this_year = service.pto_taken_this_year
+        flash.now[:notice] = t("flash.successfully_created")
+        render turbo_stream: [
+          turbo_stream.replace("pagination", partial: "shared/pagination", locals: {records: @time_requests}),
+          turbo_stream.prepend("records-list", partial: "time_requests/time_request", locals: {record: @time_request}),
+          turbo_stream.replace("time_off_balance", partial: "time_requests/time_off_balance", locals: {user_id: @time_request.user_request.user_id}),
+          turbo_stream.replace("right", partial: "shared/right"),
+          turbo_stream.replace("notification_alert", partial: "layouts/alert")
+        ]
+      else
```
Comment (MehdiElrafi) [inline app/controllers/time_requests_controller.rb:83]: Also this else you can remove it

File: app/controllers/time_requests_controller.rb [hunk 3]
```diff
@@ -89,20 +93,25 @@ def create
...
-      render turbo_stream: [
-        turbo_stream.replace(@time_request.dom_id, partial: "time_requests/time_request", locals: {record: @time_request}),
-        turbo_stream.replace("time_off_balance", partial: "time_requests/time_off_balance", locals: {user_id: @time_request.user_request.user_id}),
-        turbo_stream.replace("right", partial: "shared/right"),
-        turbo_stream.replace("notification_alert", partial: "layouts/alert")
-      ]
+    @time_request.assign_attributes(time_request_params)
+    if @time_request.valid?
+      @time_request.duration = new_time_request_duration(time_request_params, @user.id, @user.function.schedule)
+      if @time_request.update(time_request_params)
```
Comment (MehdiElrafi) [inline app/controllers/time_requests_controller.rb:100]: The same here

File: app/controllers/time_requests_controller.rb [hunk 4]
```diff
@@ -89,20 +93,25 @@ def create
...
+        @pto_awaiting_validation = service.pto_awaiting_validation
+        @pto_taken_this_year = service.pto_taken_this_year
+        flash.now[:notice] = t("flash.successfully_updated")
+        render turbo_stream: [
+          turbo_stream.replace(@time_request.dom_id, partial: "time_requests/time_request", locals: {record: @time_request}),
+          turbo_stream.replace("time_off_balance", partial: "time_requests/time_off_balance", locals: {user_id: @time_request.user_request.user_id}),
+          turbo_stream.replace("right", partial: "shared/right"),
+          turbo_stream.replace("notification_alert", partial: "layouts/alert")
+        ]
+      else
```
Comment (MehdiElrafi) [inline app/controllers/time_requests_controller.rb:112]: Also here

**Review by MehdiElrafi — CHANGES_REQUESTED** [review]

**Review by MehdiElrafi — APPROVED** [review]

**Pull request merged**